tests/core/gui/test_main_window.py

```diff
@@ -0,0 +1,194 @@
+# Parsec Cloud (https://parsec.cloud) Copyright (c) AGPLv3 2019 Scille SAS
+
+import pytest
+from PyQt5 import QtCore, QtWidgets
+
+from parsec.core.gui.lang import translate
+from parsec.core.types import BackendOrganizationFileLinkAddr
+
+
+@pytest.fixture
+async def logged_gui_with_files(
+    aqtbot, running_backend, backend, autoclose_dialog, logged_gui, bob, monkeypatch
+):
+    w_w = await logged_gui.test_switch_to_workspaces_widget()
+
+    assert logged_gui.tab_center.count() == 1
+
+    monkeypatch.setattr(
+        "parsec.core.gui.workspaces_widget.get_text_input", lambda *args, **kwargs: ("w1")
+    )
+    monkeypatch.setattr(
+        "parsec.core.gui.files_widget.get_text_input", lambda *args, **kwargs: ("dir1")
+    )
+
+    await aqtbot.mouse_click(w_w.button_add_workspace, QtCore.Qt.LeftButton)
+
+    def workspace_button_ready():
+        assert w_w.layout_workspaces.count() == 1
+        wk_button = w_w.layout_workspaces.itemAt(0).widget()
+        assert not isinstance(wk_button, QtWidgets.QLabel)
+
+    await aqtbot.wait_until(workspace_button_ready)
+
+    f_w = logged_gui.test_get_files_widget()
+    wk_button = w_w.layout_workspaces.itemAt(0).widget()
+    async with aqtbot.wait_exposed(f_w), aqtbot.wait_signal(f_w.folder_changed):
+        await aqtbot.mouse_click(wk_button, QtCore.Qt.LeftButton)
+
+    def _entry_available():
+        assert f_w.workspace_fs.get_workspace_name() == "w1"
+        assert f_w.table_files.rowCount() == 1
+
+    await aqtbot.wait_until(_entry_available)
+
+    def folder_ready():
+        assert f_w.isVisible()
+        assert f_w.table_files.rowCount() == 2
+        folder = f_w.table_files.item(1, 1)
+        assert folder
+        assert folder.text() == "dir1"
+
+    async with aqtbot.wait_signal(f_w.folder_stat_success, timeout=3000):
```

Comment:
Why the longer-than-defalt timeout here ? (it's always very fishy to have something like this...)

Reply:
In fact this entire test is pointless because there is bellow some `wait_until` assertion